src/core/util/config.cc

```diff
@@ -161,7 +161,9 @@ void Configuration::ParseConfig()
   // Help options
   if (m_Map.count("help"))
     {
-      std::cout << config_options << std::endl;
+      // Logging is not setup yet
+      core::SetupLogging(m_Map);
```

Comment:
Another option is to move logging initialization from core instance initialization into configuration initialization (e.g., in this function right after the map is notified). This way we only need to call the function once. An argument against doing so would be that the logging setup is then not decoupled from config parsing - but initialization requires a parsed map and per-instance configurations would require per-logging configurations anyway (eventually, when we're not using the global logging singleton).

Reply:
Changed it as you suggested.